core/src/main/java/org/apache/james/core/quota/QuotaLimit.java

```diff
@@ -0,0 +1,81 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.core.quota;
+
+import java.util.Objects;
+
+public class QuotaLimit {
+
+    public static QuotaLimit of(QuotaComponent component, QuotaScope scope, String identifier, QuotaType quotaType, long limit) {
+        return new QuotaLimit(component, scope, identifier, quotaType, limit);
+    }
+
+    private final QuotaComponent quotaComponent;
+    private final QuotaScope quotaScope;
+    private final String identifier;
```

Comment:
Thinking what we could do here...

Should be at least an Optional (global quota limit will not have an identifier for example)

After the identifier will be either a Username (account limit) or a Domain (domain limit)

But I guess if we want to be flexible to extensions even on the scope, String is likely fine I guess. But I would still make it Optional, for the global case

Reply:
identifier is in primary key so it could not be null. In case of global, we could set a fixed value to the identifier ("APP" for example)

Reply:
Correct! I think just "global" then